framework/core/image.h

```diff
@@ -119,6 +119,18 @@ struct ImageBuilder : public allocated::Builder<ImageBuilder, VkImageCreateInfo>
 		return *this;
 	}
 
+	ImageBuilder &with_compression(VkImageCompressionControlEXT *image_compression_control)
+	{
+		if (image_compression_control)
+		{
+			image_compression_control->pNext = create_info.pNext;
+
+			create_info.pNext = image_compression_control;
+		}
+
```

Comment:
I didn't expect you to move the compression code completely out of the image builder, but it's probably a good approach since it means that we can add lots of extension support without adding additional members that mostly get unused.

However, I would suggest that in this case. the function be changed to something more generic

```suggestion
	template<typename ExtensionType>
	ImageBuilder &with_extension(ExtensionType &extension)
	{
		extension.pNext = create_info.pNext;
		create_info.pNext = &extension;
	}
```

This still isn't perfect, since we should really ensure that we're not overwriting `extension.pNext`, but that can be fixed up later.

Reply:
Addressed in newer commit